Library/Homebrew/tap.rb

```diff
@@ -753,34 +753,32 @@ def cask_files_by_name
     end
   end
 
-  # Check whether the file has a Ruby extension.
-  sig { params(file: Pathname).returns(T::Boolean) }
-  def ruby_file?(file)
-    file.extname == ".rb"
+  sig { returns(Regexp) }
+  def formula_file_regex
+    @formula_file_regex ||= case formula_dir.basename.to_s
+    when "Formula"
+      %r{^Formula(/[^/]+)+\.rb$}
```

Comment:
should this have two `+`?

Reply:
Yep, it looks to be correct.

```rb
~ $ irb
irb(main):001:0> re = %r{^Formula(/[^/]+)+\.rb$}
=> /^Formula(\/[^\/]+)+\.rb$/
irb(main):002:0> re.match?("Formula/a/amber.rb")
=> true
irb(main):003:0> re.match?("Formula/amber.rb")
=> true
irb(main):004:0> re.match?("Formula.rb")
=> false
```

I can add some tests to make it clearer though.

Reply:
Tests were added in https://github.com/Homebrew/brew/pull/18167/commits/62a4734c9026500b77d30d092252b7e1ad59e6cc.